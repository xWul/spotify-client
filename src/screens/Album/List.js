/* LIBS */
import React from 'react'
import styled from 'styled-components'
/* COMPONENTS */
import Card from './Card'

const Container = styled.div`
  width: 100%;
  display: flex;
  flex-wrap: wrap;
`

const AlbumsList = ({ items }) => {
  return (
    <Container>
      {
        items.map(item => {
          return <Card key={item.id} album={item} />
        })
      }
    </Container>
  )
}

export default AlbumsList
