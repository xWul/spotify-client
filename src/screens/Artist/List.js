/* LIBS */
import React from 'react'
import styled from 'styled-components'
/* COMPONENTS */
import Card from './Card'

const Container = styled.div`
  width: 100%;
  display: flex;
  flex-wrap: wrap;
`

const ArtistList = ({ items = [] }) => {
  return (
    <Container>
      {
        items.map(item => {
          return <Card key={item.id} artist={item} />
        })
      }
    </Container>
  )
}

export default ArtistList
