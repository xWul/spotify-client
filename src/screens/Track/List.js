/* LIBS */
import React from 'react'
/* COMPONENTS */
import Card from './Card'

const List = ({ items }) => {
  return (
    <ul>
      {items.map(track => {
        return <Card  key={track.id} track={track} />
      })}
    </ul>
  )
}

export default List
