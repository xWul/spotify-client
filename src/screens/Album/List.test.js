import React from 'react'
import renderer from 'react-test-renderer'

import List from './List'

const MockAlbums = [
  {
    id: '0',
    name: 'Test 1',
    images: [],
    artists: ['0', '1', '2'],
    available_markets: ['BR']
  },
  {
    id: '1',
    name: 'Test 2',
    images: [{url: ''}],
    artists: ['test'],
    available_markets: ['CH']
  },
]

describe('List Albums', function () {
  test('Albums List renderer empty', () => {
    const component = renderer.create(
      <List />,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Albums List renderer with data', () => {
    const component = renderer.create(
      <List items={MockAlbums}/>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
