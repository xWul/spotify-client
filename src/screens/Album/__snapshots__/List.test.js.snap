// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Albums Albums List renderer empty 1`] = `
<div
  className="sc-VigVT eejlHa"
/>
`;

exports[`List Albums Albums List renderer with data 1`] = `
<div
  className="sc-VigVT eejlHa"
>
  <div
    className="sc-gzVnrw eVrGYh"
  >
    <div
      className="sc-htoDjs hscZmE"
      onClick={[Function]}
    />
    <div
      className="sc-gZMcBi fuJoju"
    >
      <div
        className="sc-iwsKbI khvyYm"
      >
        <h3
          className="sc-dnqmqq gogqhM"
        >
          Test 1
        </h3>
        <div
          className="sc-bwzfXH gICfuL"
        >
          <svg
            className="sc-bdVaJa dUXLKb"
            onClick={[Function]}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 21.35l-1.45-1.32c-5.15-4.67-8.55-7.75-8.55-11.53 0-3.08 2.42-5.5 5.5-5.5 1.74 0 3.41.81 4.5 2.09 1.09-1.28 2.76-2.09 4.5-2.09 3.08 0 5.5 2.42 5.5 5.5 0 3.78-3.4 6.86-8.55 11.54l-1.45 1.31z"
            />
          </svg>
        </div>
      </div>
      <p>
        Various artists
      </p>
      <span
        className="sc-gqjmRU bDpeBZ"
      >
        Available
      </span>
    </div>
  </div>
  <div
    className="sc-gzVnrw eVrGYh"
  >
    <div
      className="sc-htoDjs kLTpmI"
      onClick={[Function]}
    />
    <div
      className="sc-gZMcBi fuJoju"
    >
      <div
        className="sc-iwsKbI khvyYm"
      >
        <h3
          className="sc-dnqmqq gogqhM"
        >
          Test 2
        </h3>
        <div
          className="sc-bwzfXH gICfuL"
        >
          <svg
            className="sc-bdVaJa dUXLKb"
            onClick={[Function]}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 21.35l-1.45-1.32c-5.15-4.67-8.55-7.75-8.55-11.53 0-3.08 2.42-5.5 5.5-5.5 1.74 0 3.41.81 4.5 2.09 1.09-1.28 2.76-2.09 4.5-2.09 3.08 0 5.5 2.42 5.5 5.5 0 3.78-3.4 6.86-8.55 11.54l-1.45 1.31z"
            />
          </svg>
        </div>
      </div>
      <p>
        
      </p>
      <span
        className="sc-gqjmRU eaDRmZ"
      >
        Not Available
      </span>
    </div>
  </div>
</div>
`;
